macro/src/lib.rs

```diff
@@ -85,61 +89,60 @@ pub fn subxt(args: TokenStream, input: TokenStream) -> TokenStream {
         Err(e) => return TokenStream::from(e.write_errors()),
     };
 
-    let crate_path = match args.crate_path {
-        Some(crate_path) => crate_path.into(),
-        None => subxt_codegen::CratePath::default(),
-    };
-    let mut derives_registry = if args.no_default_derives {
-        DerivesRegistry::new()
-    } else {
-        DerivesRegistry::with_default_derives(&crate_path)
-    };
+    let mut codegen = CodegenBuilder::new();
 
-    let universal_derives = args.derive_for_all_types.unwrap_or_default();
-    let universal_attributes = args.attributes_for_all_types.unwrap_or_default();
-    derives_registry.extend_for_all(
-        universal_derives,
-        universal_attributes.iter().map(|a| a.0.clone()),
-    );
+    // Use the item module that the macro is on:
+    codegen.set_target_module(item_mod);
+
+    // Use the provided crate path:
+    if let Some(crate_path) = args.crate_path {
```

Comment:
The args used in the macro are very similar to the args used in the CLI, but I think it would be difficult/not worth it to unify them somehow?

Reply:
Yeah; I mean ideally they would be about identical. The types will vary a bit just because they come from different sources (eg strings from a terminal vs syn::Paths from a lib call or whatever)